src/main/java/org/jenkinsci/plugins/vsphere/builders/Clone.java

```diff
@@ -51,12 +53,13 @@
     private final String folder;
     private final String customizationSpec;
     private final boolean powerOn;
+    private final Integer timeoutInSeconds;
```

Comment:
I'm only just *one* step ahead of you at present - it's a "learning experience" for both of us :-)
...unfortunately, this means that I'm sometimes pointing you in the wrong direction :-(

I've just been reading https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility and that's reminded me that, when Jenkins loads the configuration off disk, it doesn't actually call the constructor at all.
This means that we can't rely on the constructor setting the field to a value before perform() gets run because, where a user has an old job (i.e. no timeoutInSeconds field), and their job runs with the new plugin code (which is looking for a timeoutInSeconds field), then perform() will find that field is set to zero.
So I think that we need to decide what we want the default to be (should it wait forever, or one minute, or what?  e.g. In my experience, one minute isn't enough for a Windows VM!), and then defer deciding what to until within the perform() method (which isn't what I'd do if I could rely on the constructor being called!).

I'm not familiar with what this code did before there was a timeout - did it wait "forever"?  If so, I'd suggest that this is what the default behavior should be as, that way, existing customers will see no difference, but can decide to take advantage of the new flexibility if they choose to.

(I do apologize - I was expecting this part to be *way* simpler than it's turning out to be!)